modules/validation/binding.go

```diff
@@ -44,12 +44,18 @@ func addGitRefNameBindingRule() {
 			}
 			// Additional rules as described at https://www.kernel.org/pub/software/scm/git/docs/git-check-ref-format.html
 			if strings.HasPrefix(str, "/") || strings.HasSuffix(str, "/") ||
-				strings.HasPrefix(str, ".") || strings.HasSuffix(str, ".") ||
-				strings.HasSuffix(str, ".lock") ||
-				strings.Contains(str, "..") || strings.Contains(str, "//") {
+				strings.HasSuffix(str, ".") || strings.Contains(str, "..") ||
+				strings.Contains(str, "//") {
 				errs.Add([]string{name}, ErrGitRefName, "GitRefName")
 				return false, errs
 			}
+			parts := strings.Split(str, "/")
```

Comment:
How about using a regex instead of doing similar checks twice? https://stackoverflow.com/a/12093994

Reply:
That won't work in golang as its regex does not support all required features. I tried that exact stackoverflow topic regex first when implementing this